### PR TITLE
Toggle rules at runtime

### PR DESCRIPTION
**Pull request by Anilm3**

### CODE REVIEW

File: src/rule.hpp [hunk 1]
```diff
@@ -103,8 +105,31 @@ class rule
     rule(const rule&) = delete;
     rule& operator=(const rule&) = delete;
 
-    rule(rule&&) = default;
-    rule& operator=(rule&&) = default;
+    // Constructors are not atomic...
```
Comment (cataphract) [inline src/rule.hpp:108]: I think the comment is a bit misleading. The problem is that the atomic is not movable.

**Review by cataphract — APPROVED** [review]

File: src/rule.hpp (hunk 1 above)
Comment (Anilm3) [reply src/rule.hpp:108]: Okay let me rephrase it to make clear that when using the move constructor there should be no expectation of the atomic behaving as such.
